### PR TITLE
Issue 30/more button styles

### LINKED ISSUE #30: Different Variant styles for Buttons
Like Bootstrap `outline`, `normal`

### PR DESCRIPTION
**Pull request by raphael-goetz**
This MR would resolve #30
- added variants to buttons: none, normal, filled, outlined
- renamed prop "variants" to "color"

### CODE REVIEW

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -9,7 +9,9 @@ import {getChild, getContent} from "../../utils/utils";
 export interface ButtonType extends DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement> {
     children: ReactNode | ReactNode[]
     //defaults to primary
-    variant?: "primary" | "secondary" | "info" | "success" | "warning" | "error",
+    color?: "primary" | "secondary" | "info" | "success" | "warning" | "error",
+    //default to normal
+    variant?: "none" | "normal" | "outlined" | "filled",
```
Comment (Taucher2003) [inline src/components/button/Button.tsx:14]: You have renamed the `variant` prop to `color`. Please make sure to update all usages of the prop as well.

Looking at the failed snapshot tests (https://gitlab.com/code0-tech/pictor/-/jobs/6194085242/artifacts/browse/__snapshots__/__diff_output__/), it looks like at least the Button/ButtonGroups, Card/Test and Dropdown/Dropdowns stories are affected.

File: src/components/button/Button.style.scss [hunk 2]
```diff
@@ -38,6 +36,36 @@
   .button--#{$name} {
     @include box(true, $color);
 
+    &.button__withoutBorder {
```
Comment (Taucher2003) [inline src/components/button/Button.style.scss:39]: Naming convention for css would be `button--without-border` (Also applies to the other classes)

But I do think that we should name the classes like their actual variants. So something like `button--none`, `button--filled` and `button--outlined`.

File: src/components/button/Button.tsx [hunk 3]
```diff
@@ -22,13 +24,20 @@ export interface ButtonIconType {
...
+    const {children, variant = "normal", color = "primary", active = false, disabled = false, ...args} = props
     const icon = getChild(children, ButtonIcon)
     const content = getContent(children, ButtonIcon)
 
+    const getVariant = (): string => {
+        if (variant == "none") return "button__withoutAll";
+        if (variant == "outlined") return "button__withoutBorder";
+        if (variant == "filled") return "button__withoutBackground";
+        return "";
+    }
```
Comment (Taucher2003) [inline src/components/button/Button.tsx:36]: This is static and does not need to be a function. What do you think about making this a mapping object?
```suggestion
    const variantMappings = {
      none: "button--without-all",
      outlined: "button--without-border",
      filled: "button--without-background"
    }
```

Then you can get the variant like this: `variantMappings[variant] ?? ""`

File: src/components/button/Button.style.scss [hunk 4]
```diff
@@ -38,6 +36,36 @@
...
+      &:active, &.button--none:active {
+        @include border();
+      }
+    }
+
+    &.button--none {
+      background: transparent;
+      border-color: transparent;
+      &:hover {
+        background: mix($bodyBg, if($color == $primary, $white, $color), if($color == $primary, 90%, 80%));
```
Comment (nicosammito) [inline src/components/button/Button.style.scss:62]: duplicated code here for background and border-color. We should outsource this into a function in helpers and reuse it everywhere we need it

**Review by nicosammito — CHANGES_REQUESTED** [review]
Great work, just a small change request 👍🏻

**Review by nicosammito — APPROVED** [review]

**Pull request merged**